### PR TITLE
- remove random sleep for decreased delay while typing

### LINKED ISSUE #530: typeString() and keyTap() slow even with 1ms delay
`robot.setKeyboardDelay(1);
robot.typeString('*cough');
`

is much slower than having *cough set in the clipboard and doing

`	robot.keyToggle('control','down')
	robot.keyTap('v');`

I think it could be the blocking nature of the module, and the fact that my PC is very slow.

### LINKED ISSUE #547: Maximum typing speed is rather slow
Why does keypress.c always sleep 62-125 ms per tap?

### PR DESCRIPTION
**Pull request by Kyusung4698**
- add microsleeps between toggleKeyCode for keyTap
- remove sleep from WIN32_KEY_EVENT_WAIT
- remove microsleep from X_KEY_EVENT_WAIT

Fixes #530
Fixes #547

### CODE REVIEW

**Comment by Kyusung4698** [comment]
Fixes #530
Fixes #547

**Comment by ricardopolo** [comment]
Please merge  this!!

File: src/keypress.c [hunk 1]
```diff
@@ -314,9 +313,9 @@ void typeStringDelayed(const char *str, const unsigned cpm)
...
-
-		if (mspc > 0) {
-			microsleep(mspc + (DEADBEEF_UNIFORM(0.0, 62.5)));
+    
+    if (mspc > 0) {
+			microsleep(mspc);
 		}
 	}
-}
+}
```
Comment (rayfoss) [inline src/keypress.c]: Please use tabs... it's a C thing. Also leave the new line at the end of the file. Thank you

**Comment by rayfoss** [comment]
Looks good aside from the whitespace. The title is a little misleading... setKeyboardDelay and typeStringDelayed still work as advertised. typeString is just much faster.

@Kyusung4698 Nicholas... please merge https://github.com/Kyusung4698/robotjs/pull/1

**Comment by Kyusung4698** [comment]
@rayfoss Hi, thanks for the whitespace fix. It's now merged.

**Comment by ricardopolo** [comment]
Now we are all waiting to @octalmage :shipit:

**Comment by Menomenta** [comment]
This merge would make my project a lot more useful, hoping it can be merged soon. :)

**Comment by rayfoss** [comment]
@octalmage would be a really cool guy if he merged this...

@Menomenta Mean while you can just fork and merge the PR's into your project...

```shell
git clone git@github.com:octalmage/robotjs.git ; cd robotjs
git remote add upstream git@github.com:rayfoss/robotjs.git
git fetch upstream master:upstream
git fetch origin pull/567/head:567 pull/568/head:568 pull/560/head:560
git checkout upstream
git merge --no-edit master 567 568 560
git branch -d 567 568 560
git push --set-upstream upstream upstream:master

mkdir MYAPP ; cd MYAPP
npm init -y
npm i git://git@github.com:rayfoss/robotjs.git#master
cd node_modules/robotjs
npm install && npm run install && npm run test && cd ../..
echo done.
```

**Comment by Menomenta** [comment]
> @Menomenta Meanwhile you can just fork and merge the PR's into your project...

I didn't know this was possible with npm, thanks!

**Comment by oktapodia** [comment]
Thank you for your PR

**Comment by mmkal** [comment]
@oktapodia @octalmage thanks for merging - can this be published to npm?

**Comment by chap-wong** [comment]
I really hope the npm can be updated as well, but it seems won't be at any time soon. I will just create a fork for my project as a placeholder. As soon as the npm is published, I will replace my fork.

**Comment by mmkal** [comment]
@chap-wong I see you created https://github.com/chap-wong/robotjs, did you ever publish to npm?

(@oktapodia @octalmage bump re publishing)

**Comment by chap-wong** [comment]
@mmkal I didn't publish it on npm, since npm can directly use github as package source.

**Comment by H-s-O** [comment]
The changes in this PR break `typeString` when trying to "type" into a fullscreen application, for example, a game.

I've been working on [this tool](https://github.com/H-s-O/tf2-botcheck) for Team Fortress 2, and while the typing is now instantaneous when testing typing into Notepad, it does not type into the game anymore.

**Pull request merged**